### PR TITLE
Real package verification on nginx, cli, apache

### PR DESCRIPTION
**Pull request by labbati**
### Description

This PR adds real package installation and verification centos, debian, alpine (improved).

With this PR we want to verify that under different installation methods the tracer is correctly installed and is at least able to send one trace.

- Alpine: both native and php images, CLI and nginx (not apache).
- Centos: via remi repo, all versions on CLI, nginx, apache.
    - Centos 6: due to difficulties building the images for centos 6 (packages repo has been discontinued), we reuse legacy method to validate only package installation.
- Debian: both via sury repo and native packages on debian buster and stretch. 

What this PR __does not__ do:
- it does not add zts tests
- it does not run unit and integrations tests using the real package

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [x] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -927,18 +997,6 @@ jobs:
           root: '.'
           paths: ['./build/packages', 'dockerfiles/verify_packages']
 
-  "package verification":
```
Comment (labbati) [inline .circleci/config.yml:930]: replaced by os specific targets, e.g. `verify_<OS>` + `verify_centos_6`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1123,24 +1178,13 @@ workflows:
...
-          package_type: deb
       - verify_alpine:
           requires: [ "package extension" ]
           matrix:
             parameters:
               php_package:
-                - php
+                - php php-fpm
               docker_image:
-                - alpine:3.7
```
Comment (labbati) [inline .circleci/config.yml:1143]: `php-fpm` is not aliased in 3.7, so we moved this verification (plus `php` which is just an alias for `php5` into the `php5 php5-fpm` matrix just below.

File: dockerfiles/verify_packages/docker-compose.yml [hunk 3]
```diff
@@ -16,13 +16,6 @@ services:
   '7.3-centos6': { <<: *base_image, image: 'datadog/dd-trace-ci:php-7.3_centos-6' }
   '7.4-centos6': { <<: *base_image, image: 'datadog/dd-trace-ci:php-7.4_centos-6' }
   '8.0-centos6': { <<: *base_image, image: 'datadog/dd-trace-ci:php-8.0_centos-6' }
-  '5.4-centos7': { <<: *base_image, image: 'datadog/docker-library:ddtrace_centos_7_php_54' }
```
Comment (labbati) [inline dockerfiles/verify_packages/docker-compose.yml:19]: These are now tested as a CI matrix

**Review by SammyK — APPROVED** [review]
It's really nice to see all those `verify_*` jobs in CI. Nice work @labbati! 👍

**Pull request merged**